packages/playwright-core/src/server/chromium/crNetworkManager.ts

```diff
@@ -353,15 +354,16 @@ export class CRNetworkManager {
       route,
       requestWillBeSentEvent,
       requestPausedEvent,
-      redirectedFrom
+      redirectedFrom,
+      headersOverride: headersOverride || null,
     });
     this._requestIdToRequest.set(requestWillBeSentEvent.requestId, request);
 
     if (requestPausedEvent) {
       // We will not receive extra info when intercepting the request.
       // Use the headers from the Fetch.requestPausedPayload and release the allHeaders()
       // right away, so that client can call it from the route handler.
-      request.request.setRawRequestHeaders(headersObjectToArray(requestPausedEvent.request.headers, '\n'));
+      request.request.setRawRequestHeaders(headersOverride ?? headersObjectToArray(requestPausedEvent.request.headers, '\n'));
```

Comment:
I think this whole `if` block might not apply to overrides, because we continue them right away, and will received extra info from the network stack at some point. Should we follow up?

Reply:
We already handle overridden headers in that case in Request.headers/rawRequestHeaders